pippy/PipelineSchedule.py

```diff
@@ -131,11 +121,51 @@ def __init__(
             f"""
             finished pipeline stage init, {self.stage_id=}, {self.is_first_stage=},
             {self.is_last_stage=}, {self.num_stages=},
-            {[fwd_input.shape for fwd_input in self.fwd_inputs]},
-            {[fwd_output_grad.shape for fwd_output_grad in self.fwd_output_grads]}
             """
         )
 
+    def to(self, *args, **kwargs):
+        """
+        Move the module to a new device or data type, including the buffers.
+        """
+        super().to(*args, **kwargs)
+
+        # find the device of the underlying module and move the buffers to it if they are meta
+        device = next(self.module.parameters()).device
```

Comment:
if the `to(...)` method already passes in a device argument, would this check be redundant (or potentially anti-parallel)?

Reply:
And there is a chance that `self.module` does not contain any parameter.